### PR TITLE
Aggregate allow multiple of same stage

### PR DESCRIPTION
**Pull request by dplewis**
Closes: https://github.com/parse-community/parse-server/issues/4682

Also added Mongo 3.6 aggregates stages. (sorted aphabetically)

Will update JS SDK to take array priority over object once merged.

### CODE REVIEW

**Comment by dplewis** [comment]
@oallouch Sorry this took so long. How does it look?

**Comment by oallouch** [comment]
Don't be sorry. You rock !
Looks good, but I haven't tried it yet.
I'll do it when the sdk-js is out and tell you.

Thx again

**Comment by dplewis** [comment]
@flovilmart How does this look?

File: src/Routers/AggregateRouter.js [hunk 1]
```diff
@@ -7,57 +7,51 @@ import UsersRouter   from './UsersRouter';
 const ALLOWED_KEYS = [
   'where',
   'distinct',
-  'project',
+  'addFields',
```
Comment (flovilmart) [inline src/Routers/AggregateRouter.js]: Can we make a list of `PIPELINE_KEYS` and `BASE_KEYS` and concat them in `ALLOWED_KEYS` for readability?

File: src/Routers/AggregateRouter.js [hunk 2]
```diff
@@ -34,6 +34,8 @@ const ALLOWED_KEYS = [
   'unwind',
 ];
 
+const ALLOWED_KEYS = BASE_KEYS.concat(PIPELINE_KEYS);
```
Comment (flovilmart) [inline src/Routers/AggregateRouter.js]: This will mutate BASE_KEYS IIRC :)

Comment (dplewis) [reply src/Routers/AggregateRouter.js]: Concat docs say it doesn’t mutate

Comment (flovilmart) [reply src/Routers/AggregateRouter.js]: Ok my bad, getting a bit rusty :p spread is quite nice :)

**Pull request merged**